tests/library/browsercontext-storage-state.spec.ts

```diff
@@ -96,7 +96,7 @@ it('should round-trip through the file', async ({ contextFactory }, testInfo) =>
       openRequest.onsuccess = () => {
         const request = openRequest.result.transaction('store', 'readwrite')
             .objectStore('store')
-            .put('foo', 'bar');
+            .put({ name: 'foo', date: new Date(0) }, 'bar');
```

Comment:
I think we would also benefit from a test with `key` being an array with a binary blob inside (whatever MDN means by that), and from a test with an explicit `keyPath`. Can be a follow up.